src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -108,7 +118,12 @@ private void grantPostponement(SscsCaseData sscsCaseData, PostponementRequest po
 
     private void setHearingDateToExcludedDate(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response) {
         final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
-                .filter(h -> h.getValue().getHearingDateTime().isAfter(LocalDateTime.now()))
+                .filter(h -> {
+                    if (!isNull(h.getValue().getStart())) {
+                        return LocalDateTime.now().isBefore(h.getValue().getStart());
+                    }
+                    return false;
+                })
                 .distinct()
                 .findFirst();
 
```

Comment:
There is no need for this, just use:
`Hearing hearing = sscsCaseData.getLatestHearing();
footerService.setHearingDateAsExcludeDate(hearing, sscsCaseData)`

Reply:
have looked at this have seen that getLatestHearing() sorts it by hearingid (which may not be created in order) , then datetime etc but we only care about date which i think the code i pushed does

Reply:
Agreed. The two solutions are not equivalent. Furthermore we found that `getLatestHearing` currently has side-effects so we don't want to rely on it anyway.

We'll post a small simplification of the current code, but it won't use `getLatestHearing` and that's intended.